lib/kafka_rest_client.js

```diff
@@ -72,7 +74,8 @@ function KafkaRestClient(options, callback) {
     self.discoverTopics(function discoverTopicsCallback(err, isDone) {
         if (err || !isDone) {
             self.enable = false;
-            callback(new Error('Failed to discover topics mapping'));
+            self.scheduleConnect(1);
+            callback(); // Will not return an error since there are retries to rest proxy
```

Comment:
I think it is weird to callback here and NOT return an error. That means it is impossible for this flow to 'fail'. We should probably not invoke the callback until after 1) the retry works or 2) the max amount of retries have been done, and in this case we should return an error